skops/hub_utils/_hf_hub.py

```diff
@@ -205,14 +238,69 @@ def recursively_default_dict() -> MutableMapping:
         config["sklearn"]["example_input"] = _get_example_input(data)
         config["sklearn"]["columns"] = _get_column_names(data)
     elif "text" in task:
-        if isinstance(data, list) and all(isinstance(x, str) for x in data):
-            config["sklearn"]["example_input"] = {"data": data[:3]}
-        else:
-            raise ValueError("The data needs to be a list of strings.")
+        error_message = "The data needs to be an iterable of strings."
+        try:
+            data_head = _head(data, n=3)
+            if _is_sequence_of_strings(data_head):
+                config["sklearn"]["example_input"] = {"data": data_head}
+            else:
+                raise ValueError(error_message)
+        except TypeError as e:
+            raise ValueError(error_message) from e
```

Comment:
This part is getting rather complex and distracts from the actual work being done by `_create_config`. I wonder if it would be possible to factor this code into `_get_example_input`, so that this whole block here could be simplified to:

```python
    config["sklearn"]["example_input"] = _get_example_input(data)
    if "tabular" in task:
        config["sklearn"]["columns"] = _get_column_names(data)
```

If that's not really possible, just leave it as is.

Reply:
How about something like this?
```python
config["sklearn"]["example_input"] = _get_example_input(data, task)
if "tabular" in task:
    config["sklearn"]["columns"] = _get_column_names(data)
```
Cause this would make it easier for me, I think. Alternatively, I'm thinking of something like:
```python
if "tabular" in task:
    config["sklearn"]["example_input"] = _get_example_input_from_tabular_data(data)
    config["sklearn"]["columns"] = _get_column_names(data)
elif "text" in task:
    config["sklearn"]["example_input"] = _get_example_input_from_text_data(data)
```
although the latter would imply renaming an existing function

Reply:
> config["sklearn"]["example_input"] = _get_example_input(data, task)

Looks good to me.